src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Generic.Read.cs

```diff
@@ -1271,6 +1368,12 @@ public static IEnumerable<object[]> CustomInterfaces_Enumerables()
             yield return new object[] { typeof(IDerivedICollectionOfT<string>) };
             yield return new object[] { typeof(IDerivedIList) };
             yield return new object[] { typeof(IDerivedISetOfT<string>) };
+
+            // Only modern .NET (>= 5.0) supports IReadOnlySet<T>.
```

Comment:
Nit: redundant comment; other types that only available on higher frameworks aren't commented like this.